osidb/models.py

```diff
@@ -1344,22 +1344,36 @@ def bzsync(self, *args, bz_api_key, **kwargs):
             from apps.bbsync.save import FlawBugzillaSaver
             from collectors.bzimport.collectors import FlawCollector
 
-            # sync to Bugzilla
-            bs = FlawBugzillaSaver(self, bz_api_key)
-            flaw_instance = bs.save(old_instance)
-            kwargs[
-                "auto_timestamps"
-            ] = False  # the timestamps will be get from Bugzilla
-            kwargs["raise_validation_error"] = False  # the validations were already run
-            # save in case a new Bugzilla ID was obtained
-            flaw_instance.save(no_alerts=True, *args, **kwargs)
-
-            # fetch from Bugzilla
-            fc = FlawCollector()
-            fc.no_alerts = True
-            fc.sync_flaw(flaw_instance.bz_id)
-            # Make sure the flaw instance has the latest data
-            flaw_instance.refresh_from_db()
+            try:
+                # sync to Bugzilla
+                bs = FlawBugzillaSaver(self, bz_api_key)
+                flaw_instance = bs.save(old_instance)
+                kwargs[
+                    "auto_timestamps"
+                ] = False  # the timestamps will be get from Bugzilla
+                kwargs[
+                    "raise_validation_error"
+                ] = False  # the validations were already run
+                # save in case a new Bugzilla ID was obtained
+                flaw_instance.save(no_alerts=True, *args, **kwargs)
+
+                # fetch from Bugzilla
+                fc = FlawCollector()
+                fc.no_alerts = True
+                fc.sync_flaw(flaw_instance.bz_id)
+                # Make sure the flaw instance has the latest data
+                flaw_instance.refresh_from_db()
+
+            except Exception as e:
+                # Sync failed but if it was done async the original flaw may be saved, resulting in
+                # incosnsitent data between OSIDB and BZ.
+                logger.error(f"Error when syncing flaw {self.uuid} to Bugzilla: {e}")
+                self.alert(
+                    "bzsync_failed",
+                    "The Bugzilla sync for this flaw failed in the last save, so there may be data discrepancies.",
+                    Alert.AlertType.ERROR,
+                )
+                raise e
```

Comment:
This section of the code is run only when the synchronous sync is being executed and that is basically no longer used, in my PR which I will be raising soon, it will be used only for syncing BZ comments as we need to preserve authors of the commits.

Problem here is with asynchronous, which happens in the `if` branch of this condition block. BZsyncManager is scheduled there which basically then runs [`_perform_bzsync` method](https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/sync_manager.py#L630), so the actual logging should happen there. If any problem happens in the synchronous branch, it is not a big deal since it fails directly and user is greeted with error message.